packages/main/src/cache/persist.ts

```diff
@@ -396,6 +396,32 @@ export function delLabels({
   ).run({ accountId, boxName: box.name })
 }
 
+export function addLabels({
+  accountId,
+  box,
+  uids,
+  labels
+}: {
+  accountId: ID
+  box: { name: string }
+  uids: number[]
+  labels: string[]
+}) {
+  labels.forEach(label => {
+    db.prepare(
+      `
+      insert into message_gmail_labels (id, label, message_id)
```

Comment:
When inserting please don't provide a value for `id`. The `id` column is declared `integer primary key` which in sqlite means that the database will automatically assign an `id` when the record is created, and the database can do a better job than application code of choosing an `id` value that is not already in use.

The `select` clause here selects three values: `@accountId`, `@label`, `messages.id`, and inserts those values into the columns `id`, `label`, `message_id` in the new rows. The label and message ID assignments are correct, but we do not want to use the account ID as the `id` of the new label row.

Otherwise this query is looking good.